src/commands/run.rs

```diff
@@ -0,0 +1,163 @@
+use crate::model::runnable::*;
+use crate::Error;
+use serenity::utils::MessageBuilder;
+use std::io::ErrorKind;
+
+/// Given some stdout or stderr data, format it so that it can be rendered by discord
+fn format_output(response: String) -> String {
+    if response.len() < 1990 {
+        // Response falls within size constraints
+        return format!("```\n{}\n```", response);
+    } else {
+        // we trim to 1981 chars because [TRIMMED] is 9 chars
+        let short_repsonse = &response[0..1981]; // TODO: maybe do this in place with a mutable string
+        return format!("```{}[TRIMMED]```", short_repsonse);
+    }
+}
+
+async fn reply(
+    ctx: poise::ApplicationContext<'_, crate::Data, crate::Error>,
+    code: String,
+    reply_text: String,
+) -> Result<(), Error> {
+    let interaction = ctx.interaction.unwrap();
+    let channel = match interaction.channel_id.to_channel(&ctx.discord.http).await {
+        Ok(channel) => channel,
+        Err(why) => {
+            println!("Error getting channel: {:?}", why);
+            return Ok(());
+        }
+    };
+
+    channel
+        .id()
+        .send_message(&ctx.discord.http, |m| {
+            m.content(
+                MessageBuilder::new()
+                    .mention(&interaction.member.clone().unwrap())
+                    .push("Ran")
+                    .push(code)
+                    .push("Output")
+                    .push(reply_text)
+                    .build(),
+            )
+        })
+        .await?;
+    Ok(())
+}
+
+#[derive(Debug, poise::Modal)]
+#[allow(dead_code)] // fields only used for Debug print
+struct RunModal {
+    #[name = "Code you want to run"]
+    #[placeholder = "fn main() {\n    println!(\"Hello, world!\");\n}"]
```

Comment:
I'm curious about what might work best for editing a ~20-line function. I think the code that is part of the question would be a good placeholder? It would be really cool if a modal could feel like a mini editor.

Reply:
Good point... A few things would help with editing a 20 line function
- Having monospace support would be great, but unfortunately we can't get that with modals or slash commands just yet. The original RustBot had this feature by *not* using the command framework and just having users enter code blocks in markdown, but this is kind of a waste of time here as Discord is canning the "classic" functionality in favor of the command framework (unless you get approval from them if you *need* that functionality)
- You could inject the question code sample as the `value` of the modal instead of the placeholder, like so https://docs.rs/poise/0.2.1/src/poise/modal.rs.html#149